src/runtime-rs/crates/hypervisor/src/ch/inner_hypervisor.rs

```diff
@@ -524,7 +605,18 @@ impl CloudHypervisorInner {
 
     pub(crate) async fn capabilities(&self) -> Result<Capabilities> {
         let mut caps = Capabilities::default();
-        caps.set(CapabilityBits::FsSharingSupport);
+
+        let flags = if guest_protection_is_tdx(self.guest_protection_to_use.clone()) {
+            // TDX does not permit the use of virtio-fs.
+            CapabilityBits::BlockDeviceSupport | CapabilityBits::BlockDeviceHotplugSupport
+        } else {
+            CapabilityBits::BlockDeviceSupport
+                | CapabilityBits::BlockDeviceHotplugSupport
+                | CapabilityBits::FsSharingSupport
```

Comment:
What effect does adding the additional `CapabilityBits` have for the non-TDX case - was something broken prior to this?

Reply:
Not broken, but we only handled the default (non TDX, non-DM) cases.